batch-nft-reveal/app/components/create/Create.tsx

```diff
@@ -0,0 +1,312 @@
+import { Button, FormControl, FormErrorMessage, FormLabel, Input, Tooltip } from '@chakra-ui/react'
+import { useEthers } from '@usedapp/core';
+import { useState } from 'react'
+import { Error } from '../../components/Error'
+
+export function Create(): JSX.Element {
+  const { account } = useEthers()
+
+  const [hasError, setHasError] = useState(false);
+
+  const [tokenName, setTokenName] = useState("");
+  const [symbol, setSymbol] = useState("");
+  const [maxSupplay, setMaxSupply] = useState("");
+  const [mintCost, setMintCost] = useState("");
+  const [revealBatchSize, setRevealBatchSize] = useState("");
+  const [revealInterval, setRevealInterval] = useState("");
+  const [vrfCoordinatorV2, setVrfCoordinatorV2] = useState("");
+  const [vrfSubscriptionId, setVrfSubscriptionId] = useState("");
+  const [vrfGasLane, setVrfGasLane] = useState("");
+  const [vrfCallbackGasLimit, setVrfCallbackGasLimit] = useState("");
+
+  const [tokenNameEntered, setTokenNameEntered] = useState(false);
+  const [symbolEntered, setSymbolEntered] = useState(false);
+  const [maxSupplayEntered, setMaxSupplyEntered] = useState(false);
+  const [mintCostEntered, setMintCostEntered] = useState(false);
+  const [revealBatchSizeEntered, setRevealBatchSizeEntered] = useState(false);
+  const [revealIntervalEntered, setRevealIntervalEntered] = useState(false);
+  const [vrfCoordinatorV2Entered, setVrfCoordinatorV2Entered] = useState(false);
+  const [vrfSubscriptionIdEntered, setVrfSubscriptionIdEntered] = useState(false);
+  const [vrfGasLaneEntered, setVrfGasLaneEntered] = useState(false);
+  const [vrfCallbackGasLimitEntered, setVrfCallbackGasLimitEntered] = useState(false);
+
+  const sendRequest = async () => {
+    // TODO: call function for deploy contract here
+    console.log(`Send create request with params: ${tokenName}, ${symbol}, ${maxSupplay}, ${mintCost}, ${revealBatchSize},
+    ${revealInterval}, ${vrfCoordinatorV2}, ${vrfSubscriptionId}, ${vrfGasLane}, ${vrfCallbackGasLimit}`);
+  }
+
+  const isNotNumber = (value) => {
+    return !/^[0-9]*$/.test(value);
+  }
+
+  const isNotAddress = (value) => {
+    return !/^0x[a-fA-F0-9]{40}$/.test(value);
+  }
+
+  const isNotBytes32 = (value) => {
+    return !/^0x[a-fA-F0-9]{64}$/.test(value);
+  }
+
+  const isInvalidName = !tokenName;
+  const isInvalidSymbol = !symbol;
+  const isInvalidMaxSupplay = isNotNumber(maxSupplay) || Number(maxSupplay) <= 0;
+  const isInvalidMintCost = isNotNumber(mintCost); // TODO now is in satoshi... maybe make it fractional number
```

Comment:
There is no satoshi unit on Ethereum. The cost is in ETH, as you can see in the mockup.